src/components/Context/injectWeb.js

```diff
@@ -12,14 +12,20 @@ governing permissions and limitations under the License.
 
 import { deepAssign } from "../../utils";
 
+// URLs reported by browser APIs may not be URI-compliant because browsers are more lenient
+// than the URI RFC (https://tools.ietf.org/html/rfc3986). The XDM fields we're populating,
+// however, should conform to the URI spec. For this reason, we need to encode the URL.
+// We decode the URL first so it doesn't end up double encoded if it was previously encoded.
+const encode = url => encodeURI(decodeURI(url));
```

Comment:
So when/if another decodeURI were to be applied on the result (e.g. decodeURI(encodeURI(decodeURI(url)))) we're sure that the end result would end up being the original url?